MantidQt/CustomInterfaces/test/ReflDataProcessorPresenterTest.h

```diff
@@ -0,0 +1,633 @@
+#ifndef MANTID_CUSTOMINTERFACES_REFLDATAPROCESSORPRESENTERTEST_H
+#define MANTID_CUSTOMINTERFACES_REFLDATAPROCESSORPRESENTERTEST_H
+
+#include <cxxtest/TestSuite.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "MantidAPI/FrameworkManager.h"
+#include "MantidAPI/TableRow.h"
+#include "MantidDataObjects/EventWorkspace.h"
+#include "MantidQtCustomInterfaces/Reflectometry/ReflGenericDataProcessorPresenterFactory.h"
+#include "MantidQtMantidWidgets/DataProcessorUI/DataProcessorMockObjects.h"
+#include "MantidQtMantidWidgets/DataProcessorUI/GenericDataProcessorPresenter.h"
+#include "MantidQtMantidWidgets/DataProcessorUI/ProgressableViewMockObject.h"
+#include "MantidTestHelpers/WorkspaceCreationHelper.h"
+
+using namespace Mantid::API;
+using namespace Mantid::Kernel;
+using namespace Mantid::DataObjects;
+using namespace MantidQt::CustomInterfaces;
+using namespace MantidQt::MantidWidgets;
+using namespace testing;
+
+class ReflDataProcessorPresenterTest : public CxxTest::TestSuite {
+
+private:
+  ITableWorkspace_sptr
+  createWorkspace(const std::string &wsName,
+                  const DataProcessorWhiteList &whitelist) {
+    ITableWorkspace_sptr ws = WorkspaceFactory::Instance().createTable();
+
+    const int ncols = static_cast<int>(whitelist.size());
+
+    auto colGroup = ws->addColumn("str", "Group");
+    colGroup->setPlotType(0);
+
+    for (int col = 0; col < ncols; col++) {
+      auto column = ws->addColumn("str", whitelist.colNameFromColIndex(col));
+      column->setPlotType(0);
+    }
+
+    if (wsName.length() > 0)
+      AnalysisDataService::Instance().addOrReplace(wsName, ws);
+
+    return ws;
+  }
+
+  ITableWorkspace_sptr
+  createPrefilledWorkspace(const std::string &wsName,
+                           const DataProcessorWhiteList &whitelist) {
+    auto ws = createWorkspace(wsName, whitelist);
+    TableRow row = ws->appendRow();
+    row << "0"
+        << "13460"
+        << "0.7"
+        << "13463,13464"
+        << "0.01"
+        << "0.06"
+        << "0.04"
+        << "1"
+        << "";
+    row = ws->appendRow();
+    row << "0"
+        << "13462"
+        << "2.3"
+        << "13463,13464"
+        << "0.035"
+        << "0.3"
+        << "0.04"
+        << "1"
+        << "";
+    row = ws->appendRow();
+    row << "1"
+        << "13469"
+        << "0.7"
+        << "13463,13464"
+        << "0.01"
+        << "0.06"
+        << "0.04"
+        << "1"
+        << "";
+    row = ws->appendRow();
+    row << "1"
+        << "13470"
+        << "2.3"
+        << "13463,13464"
+        << "0.01"
+        << "0.06"
+        << "0.04"
+        << "1"
+        << "";
+    return ws;
+  }
```

Comment:
For me, it would be easier to read something like this:

  ITableWorkspace_sptr
  createPrefilledWorkspace(const std::string &wsName,
                           const DataProcessorWhiteList &whitelist) {
    auto ws = createWorkspace(wsName, whitelist);
    std::vector<int> col1{0, 0, 1, 1};
    std::vector<int> col2{13460, 13462, 13469, 13470};
    std::vector<float> col3{0.7, 2.3, 0.7, 2.3};
    const std::string col4{"13463,13464"};
    std::vector<float> col5{0.01, 0.035, 0.01, 0.01};
    std::vector<float> col6{0.06, 0.3, 0.06, 0.06};
    const std::string col7{"0.04"};
    const std::string col8{"1"};
    const std::string col9{""};
    for (int i=0; i<5; ++i){
      TableRow row = ws->appendRow();
      row << std::to_string(col1[i])
          << std::to_string(col2[i])
          << std::to_string(col3[i])
          << col4
          << std::to_string(col5[i])
          << std::to_string(col6[i])
          << col7
          << col8
          << col9;
    }
    return ws;
  }

col1, col2, ... can have a mode meaningful name. It is shorter and  reading a row of columns is easier than reading across lines.